compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1588,6 +1588,14 @@ impl<'a, 'tcx> InferBorrowKind<'a, 'tcx> {
 impl<'a, 'tcx> euv::Delegate<'tcx> for InferBorrowKind<'a, 'tcx> {
     fn fake_read(&mut self, place: Place<'tcx>, cause: FakeReadCause, diag_expr_id: hir::HirId) {
         if let PlaceBase::Upvar(_) = place.base {
+            // We need to restrict Fake Read precision to avoid fake reading unsafe code,
```

Comment:
To test this is working as expected, you'd have to go remove the feature gaurd that was added in quick fix PR

Reply:
It's not there. The quick PR hasn't been merged yet. I need to wait for it to merge before I can do that: https://github.com/rust-lang/rust/pull/85564. So it works:

Reply:
Okay, just open this I suppose and leave a message saying if that gets merged, then that change needs to be reverted